utils/utils.go

```diff
@@ -0,0 +1,356 @@
+			break
+		}
+	}
+
+	require.NotEmpty(t, nodes, "The Nodes list returned should not be empty")
+	require.Equal(t, desiredNodeCount, len(nodes.Items),
+		"%d out of %d desired worker nodes are instantiated and running", len(nodes.Items), desiredNodeCount)
+
+	// Validate each node has a "Ready" status.
+	i, readyCount := 0, 0
+	nodeReady, retries := false, MaxRetries
+	for i < len(nodes.Items) && retries > 0 {
+		node := nodes.Items[i]
+		n, err := clientset.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+		if err != nil {
+			fmt.Printf("Waiting for Node '%s' to be returned. Retrying...\n", node.Name)
+			retries--
+			time.Sleep(RetryInterval * time.Second)
+			continue
+		}
+		for _, condition := range n.Status.Conditions {
+			if condition.Type == "Ready" {
+				nodeReady = condition.Status == "True"
+			}
+		}
+		if nodeReady {
+			readyCount++
+			t.Logf("Node: %s | Ready Status: %t\n", node.Name, nodeReady)
+			i++ // reset vars for next run
+			nodeReady = false
+			retries = MaxRetries
+			continue
+		}
+		retries--
+		if retries == 0 {
+			t.Logf("Node: %s | Ready Status: %t\n", node.Name, nodeReady)
+			i++ // reset vars for next run
+			nodeReady = false
+			retries = MaxRetries
+		} else {
+			fmt.Printf("Waiting for Node '%s' to be ready. Retrying...\n", node.Name)
+			time.Sleep(RetryInterval * time.Second)
+		}
+	}
+	require.Equal(t, readyCount, len(nodes.Items),
+		"%d out of %d Nodes are ready", readyCount, len(nodes.Items))
+
+	t.Logf("%d out of %d Nodes are ready", readyCount, len(nodes.Items))
+	fmt.Printf("%d out of %d Nodes are ready\n", readyCount, len(nodes.Items))
+}
+
+// AssertAllPodsReady ensures all pods have a "Running" or "Succeeded" status
+// phase, and a "Ready" status condition.
+func AssertAllPodsReady(t *testing.T, clientset *kubernetes.Clientset) {
+	var pods *v1.PodList
+	var err error
+
+	// Assume first non-error return of a list of pods is correct, as we
+	// currently do not know how many pods to anticipate in the cluster before
+	// it has been stood up.
+	retries := MaxRetries
+	for retries > 0 {
+		pods, err = clientset.CoreV1().Pods("").List(metav1.ListOptions{})
+		if err != nil {
+			fmt.Printf("Waiting for the list of all Pods to be returned. Retrying...\n")
+			retries--
+			time.Sleep(RetryInterval * time.Second)
+		} else {
+			break
+		}
+	}
+
+	require.NotEmpty(t, pods, "The Pods list returned should not be empty")
+
+	// Validate each pod has a "Running" or "Succeeded" status phase,
+	// and has a "Ready" status condition.
+	i, readyCount := 0, 0
+	podReady, retries := false, MaxRetries
+	for i < len(pods.Items) && retries > 0 {
+		pod := pods.Items[i]
+		p, err := clientset.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
+		if err != nil {
+			retries--
+			fmt.Printf("Waiting for Pod '%s' to be returned. Retrying...\n", pod.Name)
+			time.Sleep(RetryInterval * time.Second)
+			continue
+		}
+		if p.Status.Phase == "Running" || p.Status.Phase == "Succeeded" {
```

Comment:
You can use consts from the same file I linked earlier.

Reply:
Done